src/components/Messages/Messages.js

```diff
@@ -23,42 +24,37 @@ class Messages extends Component {
   onListenForMessages = () => {
     this.setState({ loading: true });
 
-    this.props.firebase
+    this.unsubscribe = this.props.firebase
       .messages()
-      .orderByChild('createdAt')
-      .limitToLast(this.state.limit)
-      .on('value', snapshot => {
-        const messageObject = snapshot.val();
-
-        if (messageObject) {
-          const messageList = Object.keys(messageObject).map(key => ({
-            ...messageObject[key],
-            uid: key,
-          }));
-
-          this.setState({
-            messages: messageList,
-            loading: false,
-          });
-        } else {
-          this.setState({ messages: null, loading: false });
-        }
+      .orderBy('createdAt', 'desc')
+      .limit(this.state.limit) // firestore doesn't have limitLast, so we use combination of order desc and limit
```

Comment:
Good solution. One point where Firestore has a better API!

Reply:
Indeed we not only need `orderBy` and `limit`, but also reverse(), right? I have a hard time to run through this in theory, but can we achieve it somehow without the `reverse()` and only with the Firestore API request? I don't think so, but maybe you know something.

Reply:
For getting last ${limit} messages we need to order list by field name with `desc` mode, but in UI you would like to show messages from old to new, that's why we need to reverse it. Another solution may be when we would like to get paginated data, with `startAt` and `limit` but we should know how many messages do we have.

Reply:
Yes good point. Let's keep it this way.